gcc/rust/rust-session-manager.cc

```diff
@@ -506,6 +506,11 @@ Session::parse_file (const char *filename)
   // generate crate from parser
   auto parsed_crate = parser.parse_crate ();
 
+  // If -fsyntax-only was passed, we can just skip the remaining passes.
+  // Parsing errors are already emitted in `parse_crate()`
```

Comment:
it might be worth moving this beflow the point where we dump the last so its still possible to get an ast dump

Reply:
Done in [e30d07d](https://github.com/Rust-GCC/gccrs/pull/1035/commits/e30d07de5c89e0f8a4d8304846e21f8e775c9f79)